templates/terraform/.github/workflows/auto-readme.yml

```diff
@@ -15,21 +17,35 @@ on:
 
 jobs:
   update:
-    if: github.event_name == 'schedule'
+    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
 
+    - name: Find default branch name
+      id: defaultBranch
+      shell: bash
+      env:
+        GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+      run: |
+        default_branch=$(gh repo view --json defaultBranchRef --jq .defaultBranchRef.name)
+        printf "::set-output name=defaultBranch::%s\n" "${default_branch}"
+        printf "defaultBranchRef.name=%s\n" "${default_branch}"
```

Comment:
Why are we just not using https://github.blog/changelog/2020-07-22-github-actions-better-support-for-alternative-default-branch-names/

There's native support for this in GHA.

Reply:
`$default-branch` is a macro expanded when copying a template "starter workflow" to a destination repository. It does not work with our make-based file copies. From [the documentation](https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization#creating-a-starter-workflow):
> If you need to refer to a repository's default branch, you can use the $default-branch placeholder. _When a workflow is created the placeholder will be automatically replaced with the name of the repository's default branch_. (Emphasis added.)

It does not work inside a deployed workflow.

Reply:
We could possibly use `${{ github.event.repository.default_branch }}` but the way I read the [documentation of the "schedule" event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) is that `github.event` is not available when a scheduled run takes place.

Reply:
Ok, let's stick with what you have....

Posting this for reference https://stackoverflow.com/questions/64781462/github-actions-default-branch-variable

Probably should just create our own action for this. I kind of like the API approach in the last example ☝️